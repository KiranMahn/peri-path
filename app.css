.app {
    text-align: center;
    font-size: 1.2em;
  
  }
  
  .App-logo {
    height: 40vmin;
    pointer-events: none;
  }
  
  @media (prefers-reduced-motion: no-preference) {
    .App-logo {
      animation: App-logo-spin infinite 20s linear;
    }
  }
  
  .App-header {
    background-color: #282c34;
    min-height: 100vh;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    font-size: calc(10px + 2vmin);
    color: white;
  }
  
  .App-link {
    color: #61dafb;
  }
  
  @keyframes App-logo-spin {
    from {
      transform: rotate(0deg);
    }
    to {
      transform: rotate(360deg);
    }
  }
  
  button {
    border-radius: 0.5em;
    border: none;
    padding: 0.5em;
    margin: 0.25em;
    font-size: medium;
    background-color: #009688;
    color: white;
    font-weight: bold;
  }
  
  input {
    border-radius: 0.5em;
    border: none;
    padding: 0.5em;
    background-color: rgb(244, 243, 243);
    margin: 0.5em;
  }
  
  body {
    font-family:'Trebuchet MS', 'Lucida Sans Unicode', 'Lucida Grande', 'Lucida Sans', Arial, sans-serif;
  
  }
  
  .nav-button {
    display: flex;
    flex-direction: column;
    align-items: center;
    background-color: #F8F8F8;
    color: #009688;
    border: none;
    border-radius: 0.5em;
    /* padding: 0.5em;
    margin: 0.5em; */
    font-size: medium;
    /* font-weight: bold; */
  }
  
  /* Large icons and images */
  .large-icons img, .large-icons svg {
    width: 50px;
    height: 50px;
  }
  
  /* Large text */
  .large-text {
    font-size: x-large;
  }
  
  /* High contrast colors */
  .high-contrast {
    color: #fff;
    background-color: #000;
  }
  
  .high-contrast button {
    color: #000;
    background-color: #fff;
    border: 2px solid #fff;
  }
  
  