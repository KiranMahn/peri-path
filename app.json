{
  "expo": {
    "name": "peri-path",
    "slug": "peri-path",
    "version": "1.0.0",
    "orientation": "portrait",
    "icon": "./assets/images/icon.png",
    "scheme": "myapp",
    "userInterfaceStyle": "automatic",
    "newArchEnabled": true,
    "ios": {
      "supportsTablet": true,
      "bundleIdentifier": "com.kirni.peripath",
      "infoPlist": {
        "ITSAppUsesNonExemptEncryption": false
      }
    },
    "android": {
      "adaptiveIcon": {
        "foregroundImage": "./assets/images/adaptive-icon.png",
        "backgroundColor": "#ffffff"
      }
    },
    "web": {
      "bundler": "metro",
      "output": "static",
      "favicon": "./assets/images/favicon.png"
    },
    "plugins": [
      "expo-router",
      [
        "expo-splash-screen",
        {
          "image": "./assets/images/splash-icon.png",
          "imageWidth": 200,
          "resizeMode": "contain",
          "backgroundColor": "#ffffff"
        }
      ]
    ],
    "experiments": {
      "typedRoutes": true
    },
    "extra": {
      "router": {
        "origin": false
      },
      "eas": {
        "projectId": "b0269b84-1a71-4507-812b-27e647c3c50b"
      }
    }
  }
}
