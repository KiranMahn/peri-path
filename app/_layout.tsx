import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import Home from './(tabs)/home';
import Calendar from './(tabs)/calendar';
import Track from './(tabs)/track';
import Analysis from './(tabs)/analysis';
import Profile from './(tabs)/profile';
import Learn from './(tabs)/learn';
const Stack = createStackNavigator();

const App = () => {
    return (
      <Stack.Navigator initialRouteName="Home">
        <Stack.Screen 
          name="Home" 
          component={Home} 
          options={{
            title: 'Home',
            headerStyle: {
              backgroundColor: '#rgb(0, 150, 136);',
            },
            headerTintColor: '#fff',
            headerTitleStyle: {
              fontWeight: 'bold',
            },
            headerTitleAlign: 'center',
          }}
        />
        <Stack.Screen 
            name="Calendar" 
            component={Calendar} 
            options={{
              title: 'Calendar',
              headerStyle: {
                backgroundColor: '#rgb(0, 150, 136);',
              },
              headerTintColor: '#fff',
              headerTitleStyle: {
                fontWeight: 'bold',
              },
              headerTitleAlign: 'center',
            }}
          />
          <Stack.Screen 
          name="Learn" 
          component={Learn} 
          options={{
            title: 'Learn',
            headerStyle: {
              backgroundColor: '#rgb(0, 150, 136);',
            },
            headerTintColor: '#fff',
            headerTitleStyle: {
              fontWeight: 'bold',
            },
            headerTitleAlign: 'center',
          }}
        />
        <Stack.Screen 
          name="Track" 
          component={Track} 
          options={{
            title: 'Track',
            headerStyle: {
              backgroundColor: '#rgb(0, 150, 136);',
            },
            headerTintColor: '#fff',
            headerTitleStyle: {
              fontWeight: 'bold',
            },
            headerTitleAlign: 'center',
          }}
        />
        <Stack.Screen 
          name="Analysis" 
          component={Analysis} 
          options={{
            title: 'Analysis',
            headerStyle: {
              backgroundColor: '#rgb(0, 150, 136);',
            },
            headerTintColor: '#fff',
            headerTitleStyle: {
              fontWeight: 'bold',
            },
            headerTitleAlign: 'center',
          }}
        />
        <Stack.Screen 
          name="Profile" 
          component={Profile} 
          options={{
            title: 'Profile',
            headerStyle: {
              backgroundColor: '#rgb(0, 150, 136);',
            },
            headerTintColor: '#fff',
            headerTitleStyle: {
              fontWeight: 'bold',
            },
            headerTitleAlign: 'center',
          }}
        />
      </Stack.Navigator>
    );
};

export default App;