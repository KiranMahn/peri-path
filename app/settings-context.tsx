import React, { createContext, useState, useEffect, ReactNode } from 'react';
import AsyncStorage from '@react-native-async-storage/async-storage';

interface Settings {
    largeIcons: boolean;
    largeText: boolean;
    highContrast: boolean;
}

interface SettingsContextType {
    settings: Settings;
    saveSettings: (newSettings: Settings) => void;
}

export const SettingsContext = createContext<SettingsContextType | undefined>(undefined);

interface SettingsProviderProps {
    children: ReactNode;
}

export const SettingsProvider: React.FC<SettingsProviderProps> = ({ children }) => {
    const [settings, setSettings] = useState<Settings>({
        largeIcons: false,
        largeText: false,
        highContrast: false,
    });

    useEffect(() => {
        const loadSettings = async () => {
            try {
                const savedSettings = await AsyncStorage.getItem('settings');
                if (savedSettings) {
                    setSettings(JSON.parse(savedSettings));
                }
            } catch (error) {
                console.error('Failed to load settings:', error);
            }
        };
        loadSettings();
    }, []);

    const saveSettings = async (newSettings: Settings) => {
        try {
            setSettings(newSettings);
            await AsyncStorage.setItem('settings', JSON.stringify(newSettings));
        } catch (error) {
            console.error('Failed to save settings:', error);
        }
    };

    return (
        <SettingsContext.Provider value={{ settings, saveSettings }}>
            {children}
        </SettingsContext.Provider>
    );
};
