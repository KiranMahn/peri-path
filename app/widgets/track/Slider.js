import React, { useState, useEffect, useContext } from 'react';
import { View, Text, StyleSheet } from 'react-native';
import Slider from '@react-native-community/slider';
import { SettingsContext } from '../../settings-context'; 

// stylised slider for track 

const SliderComponent = ({ sliderText, onChange, value }) => {

    const [sliderValue, setSliderValue] = useState(value);
    const { settings } = useContext(SettingsContext); 

    useEffect(() => {
        setSliderValue(value);
    }, [value]);

    const handleChange = (newValue) => {
        setSliderValue(newValue);
        onChange(newValue);
    };

    return (
        <View style={styles.container}>

            <Text style={[styles.sliderText, {fontSize: settings.largeText ? 17 : 15, color: settings.highContrast ? 'white' : '#555'}]}>{sliderText}</Text>

            <View style={styles.sliderWrapper}>

                <Slider
                    style={styles.slider}
                    testID='slider'
                    minimumValue={0}
                    maximumValue={3}
                    step={1}
                    value={sliderValue}
                    onValueChange={handleChange}
                    minimumTrackTintColor="#009688"
                    maximumTrackTintColor="#ccc"
                    thumbTintColor="#009688"
                />

                <View style={styles.levels}>

                    <Text style={[styles.levelText, {fontSize: settings.largeText ? 15 : 12, color: settings.highContrast ? 'white' : '#555'}]}>None</Text>
                    <Text style={[styles.levelText, {fontSize: settings.largeText ? 15 : 12, color: settings.highContrast ? 'white' : '#555'}]}>Low</Text>
                    <Text style={[styles.levelText, {fontSize: settings.largeText ? 15 : 12, color: settings.highContrast ? 'white' : '#555'}]}>Medium</Text>
                    <Text style={[styles.levelText, {fontSize: settings.largeText ? 15 : 12, color: settings.highContrast ? 'white' : '#555'}]}>High</Text>
                    
                </View>

            </View>

        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        marginVertical: 10,
        width: '100%',
        flexDirection: 'row',
        alignItems: 'center',
    },
    sliderText: {
        marginRight: 10,
        minWidth: 100,
    },
    sliderWrapper: {
        flex: 1,
        maxWidth: '80%',
    },
    slider: {
        width: '100%',
    },
    levels: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        marginTop: 5,
    },
    levelText: {
        fontSize: 12,
        color: '#555',
    },
});

export default SliderComponent;
