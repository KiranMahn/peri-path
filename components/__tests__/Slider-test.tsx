import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import Slider from '../../app/widgets/track/Slider';
import { SettingsContext } from '../../app/settings-context'; // Import SettingsContext and its type

describe('Slider Component', () => {
  const mockOnChange = jest.fn(); // Mock function for onChange

    const renderComponent = (props: React.JSX.IntrinsicAttributes & { level: any; onTrack: any; selected: any; sliderText?: string; value?: number; onChange?: (value: number) => void; }, settings: { largeText: boolean; highContrast: boolean; largeIcons: boolean}) => {
    return render(
      <SettingsContext.Provider value={{ settings, saveSettings: jest.fn() }}>
        <Slider sliderText={undefined} onChange={undefined} value={undefined} {...props} />
      </SettingsContext.Provider>
    );
  };

  it('renders correctly with default settings', () => {
    const { getByText } = renderComponent(
      {
          sliderText: 'Test Slider', onChange: mockOnChange, value: 50,
          level: undefined,
          onTrack: undefined,
          selected: undefined
      },
      {
          largeText: false, highContrast: false,
          largeIcons: false
      }
    );

    const sliderText = getByText('Test Slider');
    expect(sliderText).toBeTruthy();
  });

  it('renders correctly with large text enabled', () => {
    const { getByText } = renderComponent(
      {
        sliderText: 'Test Slider',
        onChange: mockOnChange,
        value: 50,
        level: undefined,
        onTrack: undefined,
        selected: undefined,
      },
      {
        largeText: true,
        highContrast: false,
        largeIcons: false,
      }
    );
  
    const sliderText = getByText('Test Slider');
    const style = Array.isArray(sliderText.props.style)
      ? Object.assign({}, ...sliderText.props.style) // Flatten the array of styles
      : sliderText.props.style;
  
    expect(style).toMatchObject({ fontSize: 17 }); // Check for the specific style property
  });

  it('renders correctly with high contrast mode enabled', () => {
    const { getByText } = renderComponent(
      {
        sliderText: 'Test Slider',
        onChange: mockOnChange,
        value: 50,
        level: undefined,
        onTrack: undefined,
        selected: undefined,
      },
      {
        largeText: false,
        highContrast: true,
        largeIcons: false,
      }
    );
  
    const sliderText = getByText('Test Slider');
    const style = Array.isArray(sliderText.props.style)
      ? Object.assign({}, ...sliderText.props.style) // Flatten the array of styles
      : sliderText.props.style;
  
    expect(style).toMatchObject({ color: 'white' }); // Check for the specific style property
  });

  it('calls onChange when slider value changes', () => {
    const { getByTestId } = renderComponent(
      {
          sliderText: 'Test Slider', onChange: mockOnChange, value: 50,
          level: undefined,
          onTrack: undefined,
          selected: undefined
      },
      {
          largeText: false, highContrast: false,
          largeIcons: false
      }
    );

    const slider = getByTestId('slider'); // Assuming the slider has a testID="slider"
    fireEvent(slider, 'valueChange', 75); // Simulate slider value change

    expect(mockOnChange).toHaveBeenCalledWith(75);
  });

  it('renders correctly with a specific value', () => {
    const { getByTestId } = renderComponent(
      {
          sliderText: 'Test Slider', onChange: mockOnChange, value: 2,
          level: undefined,
          onTrack: undefined,
          selected: undefined
      },
      {
          largeText: false, highContrast: false,
          largeIcons: false
      }
    );

    const slider = getByTestId('slider');
    expect(slider.props.value).toBe(2);
  });
});