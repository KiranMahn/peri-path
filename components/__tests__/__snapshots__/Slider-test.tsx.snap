// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "marginVertical": 10,
      "width": "100%",
    }
  }
>
  <Text
    style={
      [
        {
          "marginRight": 10,
          "minWidth": 100,
        },
        {
          "color": "#555",
          "fontSize": 15,
        },
      ]
    }
  >
    Test Slider
  </Text>
  <View
    style={
      {
        "flex": 1,
        "maxWidth": "80%",
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        [
          [
            {
              "width": "100%",
            },
            {
              "height": 40,
            },
          ],
          {
            "justifyContent": "center",
          },
        ]
      }
    >
      <RNCSlider
        disabled={false}
        inverted={false}
        lowerLimit={-9007199254740991}
        maximumTrackTintColor="#ccc"
        maximumValue={3}
        minimumTrackTintColor="#009688"
        minimumValue={0}
        onChange={[Function]}
        onRNCSliderAccessibilityAction={null}
        onRNCSliderSlidingComplete={null}
        onRNCSliderSlidingStart={null}
        onRNCSliderValueChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        step={1}
        style={
          [
            {
              "width": 0,
              "zIndex": 1,
            },
            {
              "height": 40,
            },
            {
              "alignContent": "center",
              "alignItems": "center",
            },
          ]
        }
        tapToSeek={false}
        thumbTintColor="#009688"
        upperLimit={9007199254740991}
        value={50}
      />
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#555",
              "fontSize": 12,
            },
            {
              "color": "#555",
              "fontSize": 12,
            },
          ]
        }
      >
        None
      </Text>
      <Text
        style={
          [
            {
              "color": "#555",
              "fontSize": 12,
            },
            {
              "color": "#555",
              "fontSize": 12,
            },
          ]
        }
      >
        Low
      </Text>
      <Text
        style={
          [
            {
              "color": "#555",
              "fontSize": 12,
            },
            {
              "color": "#555",
              "fontSize": 12,
            },
          ]
        }
      >
        Medium
      </Text>
      <Text
        style={
          [
            {
              "color": "#555",
              "fontSize": 12,
            },
            {
              "color": "#555",
              "fontSize": 12,
            },
          ]
        }
      >
        High
      </Text>
    </View>
  </View>
</View>
`;
